plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,44 +185,97 @@ export class WinstonInstrumentation extends InstrumentationBase {
+              record = injectRecord(spanContext, record);
+            }
+            instrumentation._callHook(span, record);
 
-        if (!isSpanContextValid(spanContext)) {
-          return original.apply(this, args);
-        }
+            // Inject in metadata argument
+            let isDataInjected = false;
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                args[i] = Object.assign(args[i], record);
+                isDataInjected = true;
+                break;
+              }
+            }
+            if (!isDataInjected) {
+              const insertAt =
+                typeof args[args.length - 1] === 'function'
+                  ? args.length - 1
+                  : args.length;
 
-        for (let i = args.length - 1; i >= 0; i--) {
-          if (typeof args[i] === 'object') {
-            const record = args[i];
-            injectRecord(spanContext, record);
-            instrumentation._callHook(span, record);
-            return original.apply(this, args);
+              args.splice(insertAt, 0, record);
+            }
           }
         }
 
-        const record = injectRecord(spanContext);
+        if (!config.disableLogSending) {
+          const logRecord: Record<string, any> = {};
+          if (args.length >= 1) {
+            // Level is always first argument
+            logRecord['level'] = args[0];
 
-        const insertAt =
-          typeof args[args.length - 1] === 'function'
-            ? args.length - 1
-            : args.length;
+            // Get meta if present
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                logRecord['meta'] = args[i];
+                break;
+              }
+            }
 
-        args.splice(insertAt, 0, record);
-        instrumentation._callHook(span, record);
+            let msgArguments = 0;
+            for (let i = 1; i < args.length; i++) {
+              if (typeof args[i] === 'string') {
+                msgArguments++;
+              }
+            }
+            if (msgArguments > 0) {
+              if (msgArguments === 1) {
+                logRecord['msg'] = args[1];
+              } else {
+                // Handle string interpolation
+                const values = args.slice(2, msgArguments + 1);
+                logRecord['msg'] = util.format(args[1], ...values);
+              }
+            }
+          }
+          instrumentation._emitLogRecord(logRecord);
+        }
 
         return original.apply(this, args);
       };
     };
   }
+
+  private _emitLogRecord(record: Record<string, any>): void {
+    const { message, msg, level, meta, ...splat } = record;
+    const attributes = meta || {};
+    for (const key in splat) {
+      if (Object.prototype.hasOwnProperty.call(splat, key)) {
+        attributes[key] = splat[key];
+      }
+    }
+    const timestamp = Date.now();
```

Comment:
In the [logs API](https://github.com/open-telemetry/opentelemetry-js/blob/f665499/experimental/packages/api-logs/src/types/LogRecord.ts#L56) the comment states that timestamp should be in nanoseconds.

The spec also specifies [nanoseconds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-timestamp)

I wonder if the logs API timestamp field should be a `HrTime` instead?

Reply:
Yeah that make sense, updated to use hrTime and convert it to nanoseconds